pkgs/build-support/docker/default.nix

```diff
@@ -969,7 +969,8 @@ rec {
 
             mkdir $out
             ${if enableFakechroot then ''
-              proot -r $PWD/old_out ${bind-paths} --pwd=/ fakeroot bash -c '
+              proot -r $PWD/old_out ${bind-paths} --pwd=/ fakeroot bash -e -c '
+                if [ -e "$NIX_ATTRS_SH_FILE" ]; then . "$NIX_ATTRS_SH_FILE"; elif [ -f .attrs.sh ]; then . .attrs.sh; fi
```

Comment:
To give a bit of context: `NIX_ATTRS_*_FILE` was introduced by me in 2021 because writing attrs into `NIX_BUILD_TOP` was a little awkward for `nix-shell` use-cases where two `nix-shell` invocations in the same directory would discard each other's `.attrs.*` files. The idea was to be able to write those files into a different locations.

However, this is not the case on Nix 2.3 (I just realized that this wasn't backported, so 2.3.17 still doesn't have the relevant patch).

Now I see the following options

* bump minver to 2.4+ (I doubt this will happen)
* backport to 2.3.18 and bump minver once again
* keep the .attrs.* hack everywhere (I think we also need to add it to `runInLinuxVM` I just realized).

Damn, I should've caught the last aspect in the runInLinuxVM aspect. Perhaps I should take a break of a few days from reviewing code here :melting_face: